### PR TITLE
Implement update

### PR DESCRIPTION
**Pull request by eivindjahren**
Implements update and does some general house cleaning:

* Fixes an error in the example usage
* Moves test generators into its own file
* Moves reusable fixture into conftest.py
* Filters expected warnings in tests
* Silences a warning about np.bool in tests
* Increases the test coverage of formatted write
* Use EclArray as a member of the API-doc
* Simplify EclArray so that it can be part of the public API.

### CODE REVIEW

File: tests/test_update.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    if file_format == Format.UNFORMATTED:
+        mode = "br+"
+    else:
+        mode = "tr+"
+
+    write(filelike, data_zeros, fileformat=file_format)
+    with open(filelike, mode) as stream:
+        for entry, (kw, arr) in zip(lazy_read(stream), data):
+            assert kw.strip() == entry.read_keyword().strip()
```
Comment (pinkwah) [inline tests/test_update.py:26]: sir, please don't strip in this place of business

File: src/ecl_data_io/_unformatted/read.py [hunk 2]
```diff
@@ -44,6 +46,24 @@ def read_array(self):
 
         return array
 
+    def update(self, new_keyword=None, new_array=None):
```
Comment (pinkwah) [inline src/ecl_data_io/_unformatted/read.py:49]:
```py
def update(self, *, keyword=None, array=None):
```

File: tests/test_update.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            entry.update("NEW_NAME")
+            assert entry.read_array().tolist() == [0] * arr.size
+            assert entry.read_keyword() == "NEW_NAME"
+            entry.update(new_array=arr)
+            assert entry.read_array().tolist() == pytest.approx(arr.tolist())
+
+
+@pytest.mark.parametrize("file_format", [Format.FORMATTED, Format.UNFORMATTED])
+def test_update_bad_path(filelike, file_format):
```
Comment (pinkwah) [inline tests/test_update.py:35]: bad name

File: tests/test_update.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        for entry, (kw, arr) in zip(lazy_read(stream), data):
+            assert kw.strip() == entry.read_keyword().strip()
+            entry.update("NEW_NAME")
+            assert entry.read_array().tolist() == [0] * arr.size
+            assert entry.read_keyword() == "NEW_NAME"
+            entry.update(new_array=arr)
+            assert entry.read_array().tolist() == pytest.approx(arr.tolist())
+
+
+@pytest.mark.parametrize("file_format", [Format.FORMATTED, Format.UNFORMATTED])
```
Comment (pinkwah) [inline tests/test_update.py:34]: `file_format` -> `fileformat`

**Review by pinkwah — CHANGES_REQUESTED** [review]

File: conftest.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from os.path import join
+
+import pytest
+
+
+@pytest.fixture(params=["file", "path"])
+def filelike(request, tmp_path, tmpdir):
```
[the remark this answers is not in the document]
Comment (eivindjahren) [reply conftest.py:7]: Fixed

File: src/ecl_data_io/_unformatted/read.py (hunk 2 above)
Comment (eivindjahren) [reply src/ecl_data_io/_unformatted/read.py:49]: Fixed

File: tests/test_update.py (hunk 3 above)
Comment (eivindjahren) [reply tests/test_update.py:35]: Fixed

File: tests/test_update.py (hunk 4 above)
Comment (eivindjahren) [reply tests/test_update.py]: fixed

**Pull request merged**